src/components/RangeSlider/components/DualThumb/DualThumb.tsx

```diff
@@ -399,6 +402,32 @@ export default class DualThumb extends React.Component<Props, State> {
     }
   }
 
+  @autobind
+  private handleTrackClick(event: React.MouseEvent) {
+    const clickXPosition = this.actualXPosition(event.clientX);
+    let xPositionLower = this.props.min;
+    let xPositionUpper = this.props.max;
```

Comment:
No need for props or looking at the DOM (`getBoundingClientRect`), instead calculate the midpoint based on the current upper and lower values and determine if the relative X position is above or below and update the respective value.

Reply:
If the click happens in the selected range, you think whichever thumb is closest should be affected?

![clicks-on-track](https://user-images.githubusercontent.com/6239480/52503620-f8c30880-2b9a-11e9-8cfe-85419bbd0172.jpg)

Reply:
Yep, exactly. The UX is a bit odd if clicking near the midpoint, but we are really trying to, in the simplest way, solve for moving the thumb nearest to click. We could potentially add a kind of dead zone in the middle that does nothing when clicked, but that's us being too opinionated (in my opinion 😄)  about user intent and adds complexity which I don't think adds value.

Reply:
Great, thanks!